### ISSUE TITLE
Support of Byte basic type

### ISSUE THREAD

**Issue by harlem88** [mongodb/bson-rust#400]
According to [BSON 1.1 Spec](https://bsonspec.org/spec.html) exists a specific type to represent the `byte` basic type.

In my opinion, the addition of the byte type could also solve the forcing in this [serialization_u8](https://github.com/mongodb/bson-rust/blob/d3952d7ba90b6ba38181dfef1ff17f4beb937aef/src/ser/serde.rs#L191) where the value is forced to i32. 
The `byte` type could be used in this context and also avoid losing the reference to the starting type.

Additional info:
In case of byte serialization (`Vec<u8>`), it is converted to Bson::Array of Bson::Int32 and not to a generic BSON::Binary. 

``` rust
#[derive(Serialize)]  
struct TestBytes {
    bytes: Vec<u8>
}
```

**Comment by abr-egn** [comment]
Hi!  `byte` is not one of the BSON element types - it's a basic type in the spec because it needs to be used in the construction of other types, but it's not one of the kinds of data that a BSON value can be.  I agree that the way the spec is worded can be a little hard to follow.

For serializing an array of bytes, I'd recommend you use the `Binary` type.
